src/util/removeEmpty.js

```diff
@@ -0,0 +1,17 @@
+import { shallowEqual } from 'recompose';
+
+const isEmpty = obj => obj === '' || obj === null || shallowEqual(obj, {});
+
+const removeEmpty = object => {
+    Object.keys(object).forEach(key => {
+        if (object[key] && typeof object === 'object') {
+            removeEmpty(object[key]);
+        }
+        if (isEmpty(object[key])) {
+            delete object[key];
```

Comment:
I used `delete` because also original logic had been used. However I agree with you and changed it to functional style.